framework/src/problems/FEProblemBase.C

```diff
@@ -7140,13 +7143,16 @@ FEProblemBase::meshChangedHelper(bool intermediate_change)
 
       // Concurrent erasure from the shared hash map is not safe while we are reading from it in
       // ProjectMaterialProperties, so we handle erasure here. Moreover, erasure based on key is
-      // not thread safe in and of itself because it is a read-write operation
-      for (const auto & elem : range)
-      {
-        _material_props.eraseProperty(elem);
-        _bnd_material_props.eraseProperty(elem);
-        _neighbor_material_props.eraseProperty(elem);
-      }
+      // not thread safe in and of itself because it is a read-write operation. Note that we do not
+      // do the erasure for p-refinement because the coarse level element is the same as our active
+      // refined level element
+      if (!doingPRefinement())
```

Comment:
so it s either p or h refinement right
there is no mixing allowed

Reply:
At the moment for a given `Adaptivity::adaptMesh` you must be either h- or p-, not both. In terms of the code APIs you could `adaptMesh` with h-, and then `adaptMesh` with p, and then `adaptMesh` again with h, so on and so forth. 

In terms of an input file, though, right now you cannot mix. There is only the `switch_h_to_p_refinement` flag in the Adaptivity action that is a one-time toggle. You *can* do initial uniform h-refinement and then later p-refinement based on the `Marker` system, which is demo'd in a couple of the tests.